### PR TITLE
Add case sensitivity flag for Gene Product ID validation

### PR DESCRIPTION
**Pull request by TonyWardell**
As discussed.

### CODE REVIEW

**Comment by eddturner** [comment]
Doesn't compile -- looks like `DbXRefEntityTest` needs updating.

File: annotation-rest/src/test/resources/search.properties [hunk 1]
```diff
@@ -4,3 +4,5 @@ search.return.fields=id,geneProductId,qualifier,goId,goEvidence,ecoId,reference,
 search.field.repo2domain.map=
 search.highlight.delims=
 geneproduct.db.xref.valid.regexes=src/test/resources/DB_XREFS_ENTITIES.dat.gz
+## =============== Validation  =================
+geneproduct.db.xref.valid.casesensitive=false
```
Comment (eddturner) [inline annotation-rest/src/test/resources/search.properties]: i'd recommend adding this also to:
`annotation-rest/src/main/resources/search.properties` so it's clear this is required

File: geneproduct-rest/src/test/java/uk/ac/ebi/quickgo/geneproduct/loader/DbXRefLoaderTest.java [hunk 2]
```diff
@@ -22,14 +22,14 @@
 
 	@Test
 	public void loadFileUnsuccessfully(){
-		DbXRefLoader dbXRefLoader = new DbXRefLoader(NOWHERE_CANTFIND);
+		DbXRefLoader dbXRefLoader = new DbXRefLoader(NOWHERE_CANTFIND, false);
```
Comment (eddturner) [inline geneproduct-rest/src/test/java/uk/ac/ebi/quickgo/geneproduct/loader/DbXRefLoaderTest.java]: small one -- would be worth changing `false` to a constant that says `CASE_INSENSITIVE_MATCHING` to help readability. and same for other tests. And `true` to a constant, `CASE_SENSITIVE_MATCHING`. Ditto then for the other tests.

File: annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/service/search/SearchServiceConfig.java [hunk 3]
```diff
@@ -43,6 +43,9 @@
     @Value("${geneproduct.db.xref.valid.regexes}")
     String xrefValidationRegexFile;
 
+    @Value("${geneproduct.db.xref.valid.casesensitive}")
```
Comment (eddturner) [inline annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/service/search/SearchServiceConfig.java]: Would make sense to have a default value for whether case sensitivity is used.

```java
// something along the lines of the following:
...
private static final boolean DEFAULT_XREF_VALIDATION_IS_CASE_SENSITIVE = true;
...
@Value("${geneproduct.db.xref.valid.casesensitive:"+DEFAULT_XREF_VALIDATION_IS_CASE_SENSITIVE+"}")
```

... then if for some reason, this property is not provided in the properties file, then we'll get the default value.

Comment (TonyWardell) [inline annotation-rest/src/main/java/uk/ac/ebi/quickgo/annotation/service/search/SearchServiceConfig.java]: Got point. Done.

File: geneproduct-rest/src/test/java/uk/ac/ebi/quickgo/geneproduct/loader/DbXRefLoaderTest.java (hunk 2 above)
Comment (TonyWardell) [inline geneproduct-rest/src/test/java/uk/ac/ebi/quickgo/geneproduct/loader/DbXRefLoaderTest.java]: CASE_INSENSITIVE_MATCHING = false
makes it sound like the argument is turning in-sensitivity off.

how do you feel about 
CASE_SENSITIVE_MATCHING  = false;

Comment (eddturner) [inline geneproduct-rest/src/test/java/uk/ac/ebi/quickgo/geneproduct/loader/DbXRefLoaderTest.java]: yep 'tis fine, good good

**Pull request merged**